File: backend/src/main/kotlin/com/example/backend/models/Fournisseur.kt
package com.example.backend.models

import jakarta.persistence.*
import org.hibernate.annotations.ColumnDefault

@Entity
@Table(name = "fournisseur")
class Fournisseur {
@Id
  @GeneratedValue(strategy = GenerationType.IDENTITY) // Use IDENTITY for auto-increment
  @Column(name = "id", nullable = false)
  var id: Int? = null

  @Column(name = "code", length = 16)
  var code: String? = null

  @Column(name = "nom", length = 32)
  var nom: String? = null

  @Column(name = "statut", length = 32)
  var statut: String? = null

  @Column(name = "codepostal", length = 20)
  var codepostal: String? = null

  @Column(name = "adresse", length = 32)
  var adresse: String? = null

  @Column(name = "telephone", length = 32)
  var telephone: String? = null

  @Column(name = "email", length = 32)
  var email: String? = null

  @ColumnDefault("0")
  @Column(name = "supprimer")
  var supprimer: Int? = null
}

File: backend_py/pharmaPython/app/models/fournisseur.py
from pydantic import BaseModel
from sqlalchemy import Column, Integer, String
from app.core.db import Base

class Fournisseur(Base):
  __tablename__ = "fournisseur"

  id = Column(Integer, primary_key=True)
  code = Column(String(16))
  nom = Column(String(32))
  statut = Column(String(32))
  codepostal = Column(String(20))
  adresse = Column(String(32))
  telephone = Column(String(32))
  email = Column(String(32))
  supprimer = Column(Integer, default=0)


class FournisseurSchema(BaseModel):

  id : int
  code : str
  nom : str
  statut : str
  codepostal : str
  adresse : str
  telephone : str
  email : str
  supprimer : int=0


class FournisseurIn(BaseModel):

  code : str
  nom : str
  statut : str
  codepostal : str
  adresse : str
  telephone : str
  email : str
